File: rbz-tasks/TaskSystem/TaskSystem.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace TaskSystem
{
    public interface ITaskProcessor
    {
        void Process();

        string GetName();
    }

    public class TaskProcessor
    {
        private readonly List<ITaskProcessor> _tasks = new List<ITaskProcessor>();
        
        public void Process()
        {
            foreach (var task in _tasks)
            {
                var name = task.GetName();
                Console.WriteLine($"Running Task[{name}]\n\n\n");
                task.Process();
                Console.WriteLine($"\n\n\nStopping Task[{name}]");
            }
        }

        public void AddTask(ITaskProcessor task)
        {
            _tasks.Add(task);
        }
    }
}

File: Kotlin/rbz-task/src/main/kotlin/de/rbztechnik/task/TaskSystem.kt
package de.rbztechnik.task

class TaskSystem {

    private val tasks = arrayListOf<TaskProcessor>()

    fun addTask(taskProcessor: TaskProcessor) {
        tasks.add(taskProcessor)
    }

    fun process() {
        tasks.forEach() { task ->
            val name = task.getName()
            println("Running task $name\n\n\n")
            task.process()
            println("\n\n\nStopping task $name")
        }
    }
}